heron/instance/src/java/com/twitter/heron/instance/bolt/BoltInstance.java

```diff
@@ -193,6 +192,8 @@ public void readTuplesAndExecute(Communicator<HeronTuples.HeronTupleSet> inQueue
           stream.getComponentName(), stream.getId()).size();
 
       // We would reuse the System.nanoTime()
+      long startOfCycle = System.nanoTime();
```

Comment:
why not let `long currentTime = System.nanoTime()` and keep the `startOfCycle` unchanged? On line 225 or 226, there's a timeout check. If we move create the `startOfCycle` here, the semantics there is changed.